tf2_web_republisher/CMakeLists.txt

```
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 3.22)
+project(tf2_web_republisher)
+
+find_package(ament_cmake REQUIRED)
+
+set(THIS_PACKAGE_INCLUDE_DEPENDS
+    builtin_interfaces
+    eigen3_cmake_module
+    rclcpp
+    rclcpp_action
+    tf2
+    tf2_eigen
+    tf2_geometry_msgs
+    tf2_msgs
+    tf2_ros
+    tf2_web_republisher_msgs)
+
+foreach(package IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
+  find_package(${package} REQUIRED)
+endforeach()
+
+add_library(tf2_web_republisher src/tf2_web_republisher.cpp)
+target_include_directories(
+  tf2_web_republisher
+  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+         $<INSTALL_INTERFACE:include>)
```

A common practice is to use PROJECT_NAME variable if the library has the same name as the project
```suggestion
add_library(${PROJECT_NAME} src/tf2_web_republisher.cpp)
target_include_directories(
  ${PROJECT_NAME}
  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
         $<INSTALL_INTERFACE:include>)
```